test.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+const test = require('ava');
+const hapi = require('hapi');
+const defer = require('p-defer');
+const plugin = require('./');
+
+test.beforeEach(t => {
+  delete global.__SENTRY__;
+  t.context.server = new hapi.Server();
+});
+
+const dsn = 'https://53039209a22b4ec1bcc296a3c9fdecd6@sentry.io/4291';
+
+test('requires a dsn', async t => {
+  const { server } = t.context;
+  const err = await t.throwsAsync(async () => {
+    await server.register({
+      plugin,
+      options: {
+        client: {},
+      },
+    });
+  }, {
+    name: 'ValidationError',
+    message: /Invalid hapi-sentry options/,
+  });
+
+  t.deepEqual(err.details.map(d => d.message), [
+    '"dsn" is required',
+  ]);
+});
+
+test('exposes the sentry client', async t => {
+  const { server } = t.context;
+
+  await server.register({
+    plugin,
+    options: {
+      client: { dsn, beforeSend: (e) => e },
+    },
+  });
+
+  t.is(typeof server.plugins['hapi-sentry'].client.captureException, 'function');
+});
+
+test('exposes a per-request scope', async t => {
+  const { server } = t.context;
+
+  server.route({
+    method: 'GET',
+    path: '/',
+    handler: async (request) => {
+      t.is(typeof request.sentryScope.setTag, 'function');
+      return null;
+    },
+  });
+
+  await server.register({
+    plugin,
+    options: {
+      client: { dsn },
+    },
+  });
+
+  await server.inject({
+    method: 'GET',
+    url: '/',
+  });
+});
+
+test('captures request errors', async t => {
+  const { server } = t.context;
+
+  server.route({
+    method: 'GET',
+    path: '/',
+    handler: async () => {
+      throw new Error('Oh no!');
+    },
+  });
+
+  const deferred = defer();
+  await server.register({
+    plugin,
+    options: {
+      client: {
+        dsn,
+        beforeSend: deferred.resolve,
+      },
+    },
+  });
+
+  await server.inject({
+    method: 'GET',
+    url: '/',
+  });
+
+  const event = await deferred.promise;
+  t.is(event.message, 'Error: Oh no!');
+  t.is(event.level, 'error');
+});
+
+test('parses request metadata', async t => {
+  const { server } = t.context;
+
+  server.route({
+    method: 'GET',
+    path: '/route',
+    handler: async () => {
+      throw new Error('Oh no!');
+    },
+  });
+
+  const deferred = defer();
+  await server.register({
+    plugin,
+    options: {
+      client: {
+        dsn,
+        beforeSend: deferred.resolve,
+      },
+    },
+  });
+
+  await server.inject({
+    method: 'GET',
+    url: '/route',
+  });
+
+  const event = await deferred.promise;
+  const { request } = event;
```

Comment:
You could directly destructure this in one line like `const { request } = await deferred.promise;`